File: src/main/java/me/andre111/d20server/scripting/variable/SelfIDVariable.java
package me.andre111.d20server.scripting.variable;

import me.andre111.d20server.scripting.Context;
import me.andre111.d20server.scripting.ScriptException;
import me.andre111.d20server.scripting.TokenFinder;

public class SelfIDVariable extends Variable {
	public SelfIDVariable(String fullName, TokenFinder tokenFinder) {
		super(fullName);
	}

	@Override
	public void set(Context context, Object value) throws ScriptException {
		throw new ScriptException("Entity id is read only!");
	}

	@Override
	public Object get(Context context) throws ScriptException {
		return context.self().id();
	}
}

File: core/common/scripting/variable/self-id-variable.js
import { Variable } from "./variable.js";

export class SelfIDVariable extends Variable {
    constructor(fullName) {
        super(fullName);
    }

    set(context, value) {
        throw new Error('Entity id is read only');
    }

    get(context) {
        if(!context.self) throw new Error('No "self" entity in this context');
        return context.self.getID();
    }
}
